js/script.js

```diff
@@ -0,0 +1,116 @@
+$(document).ready(function() {
+
+	var images = [];
+
+	var counter = 0;
+
+
+	
+
+	$('button').on('click', function(e) { 
+		var $input = $("#input").val();
+		// console.log(input)
+		e.preventDefault();
+		// alert("I work");
+		$.get("https://www.reddit.com/search.json?q", {
+		q: $input
+
+	}).done(function(data) {
+		//console.log(data)
+		images = data.data.children;
+		// console.log(images)
+
+		function filterByUrl(obj) {
+			if (obj.data.url) {
+				return true
+
+			}
+			else {
+				return false
+			}
+
+
+		}
+		var imagesFilter = images.filter(filterByUrl)
+		// console.log(imagesFilter)
+		var allImages = imagesFilter.map(function(obj) {
+			return obj.data.url
+		})
+		
+		
+		function cycleImages() {
+			$('#imagecontainer').html("<img src = " + allImages[counter] + ">");
+			console.log(allImages[counter])
+			counter++;
+			if(counter >= allImages.length) {
+			counter = 0;
+			}
+		};
+
+		var myInterval = setInterval(cycleImages, 2000);
```

Comment:
Here, your ```var myInterval``` will store the ID of the interval being set with setInterval (in this case, your cycleImages slideshow). You can pass this ID in as an argument to ```clearInterval()``` to stop the slideshow after using the stop button, like this:
```
clearInterval(myInterval);
```